src/Maestro/Client/src/MaestroApiFactory.cs

```diff
@@ -15,9 +15,10 @@ public static class MaestroApiFactory
         /// <param name="accessToken">Optional BAR token. When provided, will be used as the primary auth method.</param>
         /// <param name="federatedToken">Optional federated credentials token.</param>
         /// <param name="disableInteractiveAuth">Whether to include interactive login flows</param>
-        public static IMaestroApi GetAuthenticated(string baseUri, string? accessToken, string? federatedToken, bool disableInteractiveAuth)
+        /// <param name="cachePath">Path to the local cache</param>
+        public static IMaestroApi GetAuthenticated(string baseUri, string? accessToken, string? federatedToken, bool disableInteractiveAuth, string? cachePath = "")
```

Comment:
I am not sure about this. We already have the `disableInteractiveAuth` so if the logic of choosing the auth method gets to the point that we want the interactive log-in, we should always cache.

What is the scenario where we would do interactive + no cache?
If it's valid, maybe we should swap this boolean for an enum with values of `AllowInteractive`, `AllowInteractiveWithCache`, `DisallowInteractive` since the option of `disable=true, cachePath=anything` is not a valid combination.

Reply:
Discussed offline, decided that it would be better to have the path for the local cache required and have a default value that the caller can override if need be

Reply:
It is still there with a default of empty string

Reply:
Should be removed now